### PR TITLE
Allow list / elist to use parser to check if arguments are valid

### LINKED ISSUE #267: [PE-D] Non alphanumeric name parameter for list command
[-n NAME] should contain alphanumeric and spaces only but is allowed in list command: 

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: khiaxeng/ped#14

### PR DESCRIPTION
**Pull request by pyuxiang**
Fixes #267.

### CODE REVIEW

File: src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java [hunk 1]
```diff
@@ -68,40 +68,35 @@ public EListCommand parse(String args) throws ParseException {
...
             }
             for (String name : allNames) {
                 predicates.add(new EventNameContainsExactKeywordsPredicate(name));
             }
-
-            List<String> allDetails = argMap.getAllValues(PREFIX_REMARK);
-            if (!allDetails.isEmpty()) {
-                stringFind += "\n\u2022 Requires exact event detail: " + String.join(", ", allDetails);
+            if (!allRemarks.isEmpty()) {
+                stringFind += "\n\u2022 Requires exact event detail: " + String.join(", ", allRemarks);
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java:82]:
```suggestion
                stringFind += "\n\u2022 Requires exact event remark: " + String.join(", ", allRemarks);
```

File: src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java [hunk 2]
```diff
@@ -68,40 +68,35 @@ public EListCommand parse(String args) throws ParseException {
...
             }
             for (String name : allNames) {
                 predicates.add(new EventNameContainsKeywordsPredicate(name));
             }
-
-            List<String> allDetails = argMap.getAllValues(PREFIX_REMARK);
-            if (!allDetails.isEmpty()) {
-                stringFind += "\n\u2022 Requires partial event detail: " + String.join(", ", allDetails);
+            if (!allRemarks.isEmpty()) {
+                stringFind += "\n\u2022 Requires partial event detail: " + String.join(", ", allRemarks);
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java:96]:
```suggestion
                stringFind += "\n\u2022 Requires partial event remark: " + String.join(", ", allRemarks);
```

**Review by garyljj — APPROVED** [review]
Just need to fix standardizing to remark
then LGTM

**Review by Ellevy — APPROVED** [review]
LGTM

**Pull request merged**